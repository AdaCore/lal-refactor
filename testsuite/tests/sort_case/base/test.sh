cp ./main.orig1 ./main.adb
sort_case -P default.gpr -S main.adb -PL 13 -PC 14
echo ""
echo "--  orig1 -PL 13 -PC 14"
cat ./main.adb

cp ./main.orig2 ./main.adb
sort_case -P default.gpr -S main.adb -PL 12 -PC 13
echo ""
echo "--  orig2 -PL 12 -PC 13"
cat ./main.adb

cp ./main.orig3 ./main.adb
sort_case -P default.gpr -S main.adb -PL 12 -PC 16
echo ""
echo "--  orig3 -PL 12 -PC 16"
cat ./main.adb

cp ./main.orig1 ./main.adb
sort_case -P default.gpr -S main.adb -PL 13 -PC 14 -OD declaration
echo ""
echo "--  orig1 -PL 13 -PC 14 declare"
cat ./main.adb

cp ./main.orig3 ./main.adb
sort_case -P default.gpr -S main.adb -PL 12 -PC 14 -OD declaration
echo ""
echo "--  orig3 -PL 12 -PC 14 declare"
cat ./main.adb

cp ./main.orig4 ./main.adb
sort_case -P default.gpr -S main.adb -PL 12 -PC 16 -OD declaration
echo ""
echo "--  orig4 -PL 12 -PC 16 declare"
cat ./main.adb

cp ./main.orig5 ./main.adb
sort_case -P default.gpr -S main.adb -PL 12 -PC 15 -OD declaration
echo ""
echo "--  orig5 -PL 12 -PC 15 declare"
cat ./main.adb
