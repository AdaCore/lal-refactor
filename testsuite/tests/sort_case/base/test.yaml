title: Sort "when" in case

description: sort_case tool test

driver: shell_script
