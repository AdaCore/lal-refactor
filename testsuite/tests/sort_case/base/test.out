
--  orig1 -PL 13 -PC 14
with Ada.Text_IO;

--  Sort `when Variable_Address | AA` to `when AA | Variable_Address`
--  Sort all `when`s alphabetically

procedure Main is
   type Kind_Type is
     (Hover, Command, Endian, Variable_Address, Set_TTY, AA);

   Kind : Kind_Type := Hover;
begin
   case Kind is
      when AA | Variable_Address =>
         null;

      when Command =>
         Ada.Text_IO.Put_Line ("Command");

      when Endian =>
         Ada.Text_IO.Put_Line ("Endian");

      when Hover =>
         Ada.Text_IO.Put_Line ("Hover");

      when others =>
         null;
   end case;
end Main;

--  orig2 -PL 12 -PC 13
with Ada.Text_IO;

-- Sort identifiers `when Hover | Endian` to `when Endian | Hover` only

procedure Main is
   type Kind_Type is
     (Hover, Command, Endian, Variable_Address, Set_TTY, AA);

   Kind : Kind_Type := Hover;
begin
   case Kind is
      when AA =>
         null;

      when Command =>
         Ada.Text_IO.Put_Line ("Command");

      when Endian | Hover =>
         Ada.Text_IO.Put_Line ("Endian/Hover");

      when others =>
         null;
   end case;
end Main;

--  orig3 -PL 12 -PC 16
with Ada.Text_IO;

--  Sort only `when`s; `others` should be the last

procedure Main is
   type Kind_Type is
     (Hover, Command, Endian, Variable_Address, Set_TTY, AA);

   Kind : Kind_Type := Hover;
begin
   case Kind is
      when AA | Variable_Address =>
         null;

      when Command =>
         Ada.Text_IO.Put_Line ("Command");

      when Endian =>
         Ada.Text_IO.Put_Line ("Endian");

      when Hover =>
         Ada.Text_IO.Put_Line ("Hover");

      when others =>
         null;
   end case;
end Main;

--  orig1 -PL 13 -PC 14 declare
with Ada.Text_IO;

--  Sort `when Variable_Address | AA` to `when AA | Variable_Address`
--  Sort all `when`s alphabetically

procedure Main is
   type Kind_Type is
     (Hover, Command, Endian, Variable_Address, Set_TTY, AA);

   Kind : Kind_Type := Hover;
begin
   case Kind is
      when Hover =>
         Ada.Text_IO.Put_Line ("Hover");

      when Command =>
         Ada.Text_IO.Put_Line ("Command");

      when Endian =>
         Ada.Text_IO.Put_Line ("Endian");

      when Variable_Address | AA =>
         null;

      when others =>
         null;
   end case;
end Main;

--  orig3 -PL 12 -PC 14 declare
with Ada.Text_IO;

--  Sort only `when`s; `others` should be the last

procedure Main is
   type Kind_Type is
     (Hover, Command, Endian, Variable_Address, Set_TTY, AA);

   Kind : Kind_Type := Hover;
begin
   case Kind is
      when Hover =>
         Ada.Text_IO.Put_Line ("Hover");

      when Command =>
         Ada.Text_IO.Put_Line ("Command");

      when Endian =>
         Ada.Text_IO.Put_Line ("Endian");

      when Variable_Address | AA =>
         null;

      when Hover =>
         Ada.Text_IO.Put_Line ("Hover");
   end case;
end Main;

--  orig4 -PL 12 -PC 16 declare
with Ada.Text_IO;

procedure Main is
   type Kind_Type is
     (Hover, Command, Endian, Variable_Address, Set_TTY, AA);

   Kind : Kind_Type := Hover;
begin
   case Kind is
      when Hover =>
         Ada.Text_IO.Put_Line ("Hover");

      when Command | Variable_Address .. AA =>
         Ada.Text_IO.Put_Line ("Command");

      when Endian =>
         Ada.Text_IO.Put_Line ("Endian");

      when others =>
         null;
   end case;
end Main;

--  orig5 -PL 12 -PC 15 declare
with Ada.Text_IO;

procedure Main is
   type Kind_Type is
     (Hover, Command, Endian, Variable_Address, Set_TTY, AA);

   Kind : Kind_Type := Hover;
begin
   case Kind is
      when Hover =>
         Ada.Text_IO.Put_Line ("Hover");

      when Command | Variable_Address .. AA =>
         Ada.Text_IO.Put_Line ("Command");

      when Endian =>
         Ada.Text_IO.Put_Line ("Endian");

      when others =>
         null;
   end case;
end Main;
